### PR TITLE
Convert Microphysics `saturation_adjustment` driver to new framework

### PR DESCRIPTION
**Pull request by charleskawczynski**
# Description

Converts the microphysics example, `examples/Microphysics/ex_1_saturation_adjustment.jl` to the new framework.

It would be nice to reduce the boundary condition boilerplate. cc: @simonbyrne 

Co-authors: @trontrytel 

# For review by CLIMA developers

- [x] There are no open pull requests for this already
- [x] CLIMA developers with relevant expertise have been assigned to review this submission
- [ ] The code conforms to the [style guidelines](https://climate-machine.github.io/CLIMA/latest/CodingConventions.html) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [ ] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

### CODE REVIEW

**Comment by charleskawczynski** [comment]
@trontrytel should we split off some of these contributions into smaller PRs?

**Comment by trontrytel** [comment]
> @trontrytel should we split off some of these contributions into smaller PRs?

Probably the `varsindex` stuff that is in `VariableTemplates` could go separately. The rest should go in one PR. There are not that many changes. It's just that git gets dramatic if you change file names.

**Comment by trontrytel** [comment]
> > @trontrytel should we split off some of these contributions into smaller PRs?
> 
> Probably the `varsindex` stuff that is in `VariableTemplates` could go separately. The rest should go in one PR. There are not that many changes. It's just that git gets dramatic if you change file names.

Also, the `varsindex` is from @jkozdon and I think was already a part of some PR?

File: test/Driver/driver_test.jl [hunk 1]
```diff
@@ -115,12 +115,12 @@ function main()
     ugeo_abs = FT(7)
     vgeo_abs = FT(5.5)
     Δt = solver_config.dt
-    caₕ = ugeo_abs * (Δt / Δh) + vgeo_abs * (Δt / Δh)
+    ca_h = ugeo_abs * (Δt / Δh) + vgeo_abs * (Δt / Δh)
```
Comment (charleskawczynski) [inline test/Driver/driver_test.jl:118]: What's up with changes to driver_test.jl?

Comment (trontrytel) [reply test/Driver/driver_test.jl:118]: For me and some others the subscript did not display properly. What I was seeing before was a ca_box_with_a_questionmark_inside

**Comment by trontrytel** [comment]
This PR resolves issue #683

**Comment by charleskawczynski** [comment]
Partially addressed #491

**Pull request merged**